backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponent.java

```diff
@@ -264,8 +264,11 @@ private UUID computeNodeId(String targetObjectId) {
         UUID parentElementId = this.props.getParentElementId();
         NodeDescription nodeDescription = this.props.getNodeDescription();
         NodeContainmentKind containmentKind = this.props.getContainmentKind();
-
-        String rawIdentifier = parentElementId.toString() + containmentKind.toString() + nodeDescription.getId().toString() + targetObjectId;
+        String rawIdentifier = ""; //$NON-NLS-1$
+        if (parentElementId != null) {
+            rawIdentifier += parentElementId.toString();
+        }
+        rawIdentifier += containmentKind.toString() + nodeDescription.getId().toString() + targetObjectId;
```

Comment:
The identifier of a node should stay with the same behavior. The line of code that you have changed is one of the most important part of the refresh of the diagram. It is the cornerstone of the identity of a node and it *NEEDS* the identifier of its parent. The decisions of the `ADR-007` should not be modified without a very careful discussion involving at least @pcdavid and I and I don't see why changing the type of the diagram identifier should change the way the node identifier is created.

Reply:
Done